### PR TITLE
feat: scheduler frequency inputs

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #4113 

### Description:

Note: we made small styling changes after recording this image/video

Captura de ecrã 2023-02-15, às 14 28 54

https://user-images.githubusercontent.com/9117144/219055436-fbba0a6c-dc77-4ae3-b317-ed93e0a1d890.mp4

### CODE REVIEW

File: packages/frontend/src/components/SchedulerModals/SchedulerModalBase/index.tsx [hunk 1]
```diff
@@ -36,7 +36,10 @@ const SchedulersModalBase: FC<
                 slackQuery.data?.slackTeamName !== undefined &&
                 !slackQuery.isError &&
                 hasRequiredScopes(slackQuery.data);
-            return isValidSlack ? States.SUCCESS : States.ERROR;
+            // TODO: remove forceSuccess
```
Comment (ZeRego) [inline packages/frontend/src/components/SchedulerModals/SchedulerModalBase/index.tsx:39]: This is to allow reviewing without enabling slack.
I will remove this after approval.

**Comment by rephus** [comment]
Could we increase the size of this select a bit to show the full string ? 
image

**Comment by rephus** [comment]
is this expression not valid ? :( I guess we'll have to create multiple schedulers to cover cases like this  
image

**Comment by rephus** [comment]
Eventually we will have to convert all these UTC inputs into local time (maybe except for custom) , or at least show the time locally (on a label or something). This can be done on a separate PR

**Comment by rephus** [comment]
Not related with this PR, but should we allow schedulers without targets ? 
[Screencast from 16-02-23 11:30:44.webm](https://user-images.githubusercontent.com/1983672/219340567-5f14a551-6b73-424e-89cd-686b09da597e.webm)

**Review by rephus — CHANGES_REQUESTED** [review]
Amazing, this was a lot harder than I expected, but you did a great job. It works super well.  I didn't find anything wrong in the code.

Just fix a few changes in the UI and then we can merge it.

**Comment by ZeRego** [comment]
> This monthly select looks weird image

@rephus Can't replicate this :/

**Comment by ZeRego** [comment]
> Not related with this PR, but should we allow schedulers without targets ?
> [Screencast from 16-02-23 11:30:44.webm](https://user-images.githubusercontent.com/1983672/219340567-5f14a551-6b73-424e-89cd-686b09da597e.webm)

We need to upgrade `react-hook-form` to the latest version to be able to validate arrays. 
```rules: { minLength: 4 }```

File: packages/frontend/src/utils/fieldValidators.ts [hunk 2]
```diff
@@ -59,11 +59,22 @@ export const isValidGithubToken: FieldValidator<string> =
         }
     };
 
-const cronExpressionRegex = new RegExp(
-    /^(\*|([0-9]|1[0-9]|2[0-9]|3[0-9]|4[0-9]|5[0-9])|\*\/([0-9]|1[0-9]|2[0-9]|3[0-9]|4[0-9]|5[0-9])) (\*|([0-9]|1[0-9]|2[0-3])|\*\/([0-9]|1[0-9]|2[0-3])) (\*|([1-9]|1[0-9]|2[0-9]|3[0-1])|\*\/([1-9]|1[0-9]|2[0-9]|3[0-1])) (\*|([1-9]|1[0-2])|\*\/([1-9]|1[0-2])) (\*|([0-6])|\*\/([0-6]))$/,
+// Supports values: "1" "1,2,3" "1-3" "*/5" "*"
+const cronValueRegex = new RegExp(
```
Comment (ZeRego) [inline packages/frontend/src/utils/fieldValidators.ts:63]: Improved validation
Captura de ecrã 2023-02-16, às 17 00 35

File: packages/frontend/src/components/SchedulerModals/SchedulerModalBase/SchedulerForm.tsx [hunk 3]
```diff
@@ -220,11 +214,17 @@ const SchedulerForm: FC<
...
+                                    isAddEmailDisabled
+                                        ? undefined
+                                        : () => append({ recipients: '' })
+                                }
                                 icon={'plus'}
                                 text="Add email"
-                                disabled={
-                                    disabled || !health.data?.hasEmailClient
+                                className={
+                                    isAddEmailDisabled
```
Comment (ZeRego) [inline packages/frontend/src/components/SchedulerModals/SchedulerModalBase/SchedulerForm.tsx:225]: Fixed issue where tooltips don't close when button is disabled

**Comment by rephus** [comment]
> > This monthly select looks weird image
> 
> @rephus Can't replicate this :/

Maybe it is just my browser (firefox on linux) , but we can easily fix this by settiong `width: 150px` on numeric input 

before: 
image

after: 

image

**Comment by rephus** [comment]
When I `create a new scheduler` with any frequency, then I go to `edit` it always defaults to `weekly` , this doesn't happen if you update a scheduler 

[Screencast from 17-02-23 08:37:01.webm](https://user-images.githubusercontent.com/1983672/219580596-b673925d-3435-47ce-afc6-cf48af35bc8c.webm)

**Review by rephus — APPROVED** [review]
:man_cook: :ok_hand:

**Pull request merged**